### PR TITLE
refact(feature-test): Returning experimentKey and variationKey with feature test decisions.

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Returning experimentKey and variationKey with feature test decisions.

## Test Plan
- FSC should pass.
- Updated unit tests accordingly.

## Issues
https://optimizely.atlassian.net/browse/OASIS-6247
https://optimizely.atlassian.net/browse/OASIS-5549

### CODE REVIEW

File: pkg/optimizely/client.go [hunk 1]
```diff
@@ -170,27 +172,58 @@ func (c *OptlyClient) RemoveForcedVariation(experimentKey, userID string) (*Over
 // ActivateFeature activates a feature for a given user by getting the feature enabled status and all
 // associated variables
 func (c *OptlyClient) ActivateFeature(key string, uc entities.UserContext, disableTracking bool) (*Decision, error) {
-	enabled, variables, err := c.GetAllFeatureVariables(key, uc)
+	projectConfig, err := c.OptimizelyClient.ConfigManager.GetConfig()
```
Comment (msohailhussain) [inline pkg/optimizely/client.go:175]: this should be. c.OptimizelyConfigManager.

File: pkg/optimizely/client.go [hunk 2]
```diff
@@ -170,27 +172,63 @@ func (c *OptlyClient) RemoveForcedVariation(experimentKey, userID string) (*Over
...
+	if featureDecision, err := c.DecisionService.GetFeatureDecision(decisionContext, uc); err == nil {
+		if featureDecision.Variation != nil {
+			enabled = featureDecision.Variation.FeatureEnabled
+			variationKey = featureDecision.Variation.Key
+			experimentKey = featureDecision.Experiment.Key
+
+			if featureDecision.Source == decision.FeatureTest && !disableTracking {
+				// send impression event for feature tests
+				impressionEvent := event.CreateImpressionUserEvent(decisionContext.ProjectConfig, featureDecision.Experiment, *featureDecision.Variation, uc)
+				c.EventProcessor.ProcessEvent(impressionEvent)
```
Comment (mjc1283) [inline pkg/optimizely/client.go:201]: EventProcessor should not be used outside Go SDK

File: pkg/optimizely/client.go [hunk 3]
```diff
@@ -170,27 +172,63 @@ func (c *OptlyClient) RemoveForcedVariation(experimentKey, userID string) (*Over
...
+
+		errs := new(multierror.Error)
+		for _, v := range feature.VariableMap {
+			val := v.DefaultValue
+			if enabled {
+				if variable, ok := featureDecision.Variation.Variables[v.ID]; ok {
+					val = variable.Value
+				}
+			}
+			value, e := c.GetTypedFeatureVariableValue(val, v)
```
Comment (mjc1283) [inline pkg/optimizely/client.go:213]: What is `GetTypedFeatureVariableValue`? I can't find this in the SDK, is this something added in another branch?

Comment (mikecdavis) [reply pkg/optimizely/client.go:213]: https://github.com/optimizely/go-sdk/compare/yasir/OASIS-6247

File: go.mod [hunk 4]
```diff
@@ -10,8 +10,9 @@ require (
 	github.com/go-chi/render v1.0.1
 	github.com/go-kit/kit v0.9.0
 	github.com/google/uuid v1.1.1
+	github.com/hashicorp/go-multierror v1.1.0
 	github.com/lestrrat-go/jwx v0.9.0
-	github.com/optimizely/go-sdk v1.1.3
+	github.com/optimizely/go-sdk v1.1.3-0.20200512094102-0c730e4db4b6
```
[the remark this answers is not in the document]
Comment (pawels-optimizely) [reply go.mod:15]: when we release agent, sure - it will need to point to the released version of go sdk

**Pull request closed without merging**